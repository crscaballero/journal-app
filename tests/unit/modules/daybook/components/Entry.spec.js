import { shallowMount } from "@vue/test-utils";

import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from '../../../mock-data/test-journal-state';

describe('Entry.vue', () => {
    const mockRouter = {
        push: jest.fn()
    };
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    });
    test('It must render the component correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    test('It must redirect with click on entry-container', () => {
        const entryContainer = wrapper.find('.entry-container');
        entryContainer.trigger('click');
        expect(mockRouter.push).toHaveBeenCalled();
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        });
    });
    test('Test computed properties', () => {
        expect(wrapper.vm.day).toBe(4);
        expect(wrapper.vm.month).toBe('August');
        expect(wrapper.vm.yearDay).toBe('2022, Thursday');
    });
});
