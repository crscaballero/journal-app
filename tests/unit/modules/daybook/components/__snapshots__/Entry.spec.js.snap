// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry.vue It must render the component correctly 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">4</span><span class="mx-1 fs-5">August</span><span class="mx-2 fw-light">2022, Thursday</span></div>
  <div class="entry-description">Testing new Vue App 1</div>
</div>
`;
